src/peergos/server/util/Args.java

```diff
@@ -199,7 +217,25 @@ public Path getPeergosDir() {
         } catch (IOException ioe) {
             throw new IllegalStateException(ioe.getMessage(), ioe);
         }
+    }
 
+    /** Save all parameters to a config file, excluding environment vars
+     *
+     */
+    public void saveToFile() {
+        saveToFile(fromPeergosDir(CONFIG_FILENAME, CONFIG_FILENAME));
+    }
+
+    private void saveToFile(Path file) {
+        String text = new TreeMap<>(params).entrySet().stream()
+                .filter(e -> ! envOnly.containsKey(e.getKey()))
+                .map(e -> e.getKey() + " = " + e.getValue())
+                .collect(Collectors.joining("\n"));
+        try {
+            Files.write(file, text.getBytes(), StandardOpenOption.CREATE);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
If file already exists, should it raise a runtime ex?

Reply:
I considered that, but I want to be able to add a cmd arg (either updating an existing one, or adding a new one) and have it end up in the config file.

Reply:
On balance you're probably right, and I should just edit the config file. I'll change to only write the file if it doesn't exist (not throw).